packages/sanity-astro/src/visual-editing/visual-editing.astro

```diff
@@ -0,0 +1,57 @@
+---
+import type { VisualEditingOptions } from '@sanity/visual-editing'
+
+export interface Props extends Pick<VisualEditingOptions, 'zIndex'> {
+  enabled?: boolean
+}
+
+const {enabled, zIndex} = Astro.props
+---
+
+<sanity-astro-visual-editing enabled={enabled} data-z-index={zIndex}></sanity-astro-visual-editing>
```

Comment:
Why not 
```suggestion
<sanity-astro-visual-editing enabled={enabled} z-index={zIndex}></sanity-astro-visual-editing>
```
?

That would allow observing it:
```
static observedAttributes = ['enabled', 'z-index']
```

And handle it:
```
    attributeChangedCallback(name: string, oldValue: string | null, newValue: string | null) {
      // We don't have to guard `name` as this callback is only fired when something in `observedAttributes` changes
      if (oldValue !== newValue) {
        this.updateVisualEditing()
      }
    }
```

And simplify the render:
```

    private updateVisualEditing(enabled: boolean) {
      if (this.getAttribute("enabled") === 'true') {
        // If already setup, but an attribute changed (like changing z-index), tear down previous instances
        if(this.unsubscribe) this.teardownVisualEditing()
        this.unsubscribe = enableVisualEditing({
          zIndex: this.getAttribute("z-index") || undefined,
          refresh: () => {
            return new Promise((resolve) => {
              window.location.reload()
              resolve()
            })
          },
        })
      } else {
        this.teardownVisualEditing()
      }
    }
```

Reply:
Ok sure 👍